broker/src/main/java/org/apache/rocketmq/broker/controller/ReplicasManager.java

```diff
@@ -491,4 +532,8 @@ public List<String> getControllerAddresses() {
     public List<EpochEntry> getEpochEntries() {
         return this.haService.getEpochEntries();
     }
+
+    public List<String> getAvailableControllerAddresses() {
+        return new ArrayList<>(availableControllerAddresses.keySet());
+    }
```

Comment:
How about set ArrayList initialCapacity to prevent expansion

Reply:
This list will not expand.